craft_parts/sources/git_source.py

```diff
@@ -252,7 +259,15 @@ def _clone_new(self):
             command.extend(["--branch", self.source_tag or self.source_branch])
         if self.source_depth:
             command.extend(["--depth", str(self.source_depth)])
-        self._run(command + [self.source, self.part_src_dir])
+
+        # reformat local sources
+        if self._is_source_local():
+            command.extend([f"file://{Path(self.source).resolve()}"])
```

Comment:
We could just append to the list since it's a single element.

Reply:
Changed to `append` here and in a few other spots in the file.